src/microbe_stage/editor/upgrades/VacuoleUpgradeGUI.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+public class VacuoleUpgradeGUI : VBoxContainer, IOrganelleUpgrader
+{
+    [Export]
+    public NodePath? CompoundsPath;
+
+    [Export]
+    public NodePath IsSpecializedCheckboxPath = null!;
+
+    [Export]
+    public NodePath CompoundDescriptionPath = null!;
+
+    [Export]
+    public NodePath CompoundSelectionPath = null!;
+
+#pragma warning disable CA2213
+    private OptionButton compounds = null!;
+    private Label description = null!;
+    private CheckBox isSpecializedCheckbox = null!;
+    private VBoxContainer compoundSelection = null!;
+#pragma warning restore CA2213
+
+    private List<Compound>? shownChoices;
+
+    public override void _Ready()
+    {
+        compounds = GetNode<OptionButton>(CompoundsPath);
+        description = GetNode<Label>(CompoundDescriptionPath);
+        isSpecializedCheckbox = GetNode<CheckBox>(IsSpecializedCheckboxPath);
+        compoundSelection = GetNode<VBoxContainer>(CompoundSelectionPath);
+
+        compounds.Clear();
+    }
+
+    public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
+    {
+        shownChoices = SimulationParameters.Instance.GetAllCompounds().Values.ToList();
+
+        foreach (var compound in shownChoices)
+        {
+            if (compound.IsAgent || compound.IsEnvironmental)
+                continue;
+
+            compounds.AddItem(compound.Name);
+        }
+
+        // Select glucose by default
+        var defaultCompoundIndex =
+            shownChoices.FindIndex(c => c.InternalName == "glucose");
+
+        if (defaultCompoundIndex < 0)
+            defaultCompoundIndex = 0;
+
+        // Apply current upgrade values or defaults
+        if (organelle.Upgrades?.CustomUpgradeData is StorageComponentUpgrades configuration)
+        {
+            Compound? specialization = shownChoices.Find(c => c == configuration.SpecializationFor);
+            isSpecializedCheckbox.Pressed = specialization != null;
+
+            compounds.Selected = specialization != null ?
+                shownChoices.IndexOf(specialization) :
+                shownChoices.FindIndex(c => c.InternalName == "glucose");
```

Comment:
Shouldn't this be `defaultCompoundIndex`? to not have to find the same index again.